utilities/src/main/java/io/syndesis/qe/utils/GoogleSheetsUtils.java

```diff
@@ -0,0 +1,111 @@
+package io.syndesis.qe.utils;
+
+import com.google.api.services.sheets.v4.Sheets;
+import com.google.api.services.sheets.v4.model.ClearValuesRequest;
+import com.google.api.services.sheets.v4.model.DeleteSheetRequest;
+import com.google.api.services.sheets.v4.model.Request;
+import com.google.api.services.sheets.v4.model.BatchUpdateSpreadsheetRequest;
+import com.google.api.services.sheets.v4.model.Sheet;
+import com.google.api.services.sheets.v4.model.Spreadsheet;
+import com.google.api.services.sheets.v4.model.ValueRange;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import javax.annotation.PostConstruct;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+@Slf4j
+@Component
+public class GoogleSheetsUtils {
+    @Autowired
+    private GoogleAccounts accounts;
+    private Sheets sheets;
+    public static String TEST_SHEET_ID = "";
+    public static String DATA_TEST_SPREADSHEET = "1_OLTcj_y8NwST9KHhg8etB10xr6t3TrzaFXwW2dhpXw";
+
+
+    @PostConstruct
+    private void initSheets() {
+        if (sheets == null) {
+            sheets = accounts.getGoogleAccountForTestAccount("QE Google Sheets").sheets();
+        }
+    }
+
+    public List<String> getSpreadSheetValues() {
+        return getSpreadSheetValues(TEST_SHEET_ID, "A1:E3");
+    }
+
+    public void clearSpreadSheetValues(String range) {
+        clearSpreadSheetValues(TEST_SHEET_ID, range);
+    }
+
+    public void clearSpreadSheetValues(String id, String range) {
+        ClearValuesRequest clear = new ClearValuesRequest();
+        try {
+            Sheets.Spreadsheets.Values.Clear request = sheets.spreadsheets().values().clear(id, range, clear);
+            request.execute();
+        } catch (IOException e) {
+            e.printStackTrace();
```

Comment:
fail test instead of printing the stacktrace?

Reply:
well it's not test but utils class. So maybe better would be don't catch exception here but in test (and fail)?

Reply:
it's up to you, it would not be the first util class with fail :grin: :

```
src/main/java/io/syndesis/qe/bdd/validation/SfValidationSteps.java:        fail("Unable to connect to SalesForce");
src/main/java/io/syndesis/qe/bdd/validation/SfValidationSteps.java:            fail("Salesforce contact with last name " + name + " was not found in " + DEFAULT_WAIT_TIMEOUT/1000 + "seconds. ", e);
src/main/java/io/syndesis/qe/bdd/validation/SfValidationSteps.java:            fail("Salesforce lead with email " + email + " was not found in " + DEFAULT_WAIT_TIMEOUT/1000 + " seconds.");
src/main/java/io/syndesis/qe/bdd/validation/SfValidationSteps.java:            fail("Salesforce lead with id " + leadId + " was not deleted in " + DEFAULT_WAIT_TIMEOUT/1000 + " seconds.");
src/main/java/io/syndesis/qe/bdd/validation/SfValidationSteps.java:            fail("Salesforce email of lead with id " + leadId + " was not changed in " + DEFAULT_WAIT_TIMEOUT/1000 + " seconds.");
src/main/java/io/syndesis/qe/bdd/validation/SfValidationSteps.java:            fail("Salesforce email of lead with id " + leadId + " was not changed in " + DEFAULT_WAIT_TIMEOUT/1000 + " seconds.");
src/main/java/io/syndesis/qe/bdd/validation/OpenshiftValidationSteps.java:            fail(e.getMessage());
src/main/java/io/syndesis/qe/bdd/validation/JmsValidationSteps.java:            Assertions.fail(e.getMessage());
src/main/java/io/syndesis/qe/bdd/validation/JmsValidationSteps.java:            Assertions.fail(e.getMessage());
src/main/java/io/syndesis/qe/bdd/validation/JmsValidationSteps.java:            fail("File with name " + resourceName + " doesn't exist in the resources");
src/main/java/io/syndesis/qe/bdd/validation/MqttValidationSteps.java:            Assertions.fail("Mqtt Exception was thrown during message transfer");
src/main/java/io/syndesis/qe/bdd/validation/DbValidationSteps.java:            Assertions.fail("Error: " + ex);
src/main/java/io/syndesis/qe/bdd/validation/DbValidationSteps.java:            Assertions.fail("Error: " + ex);
src/main/java/io/syndesis/qe/bdd/validation/DbValidationSteps.java:            fail("There is no result for command: " + sql);
src/main/java/io/syndesis/qe/bdd/validation/MonitoringValidationSteps.java:            Assertions.fail("No pod found for pod name: " + podName);
src/main/java/io/syndesis/qe/bdd/CommonSteps.java:                fail((deploy ? "Syndesis wasn't initialized in time" : "Syndesis wasn't undeployed in time"));
src/main/java/io/syndesis/qe/bdd/CommonSteps.java:            fail((deploy ? "Syndesis wasn't initialized in time" : "Syndesis wasn't undeployed in time"));
src/main/java/io/syndesis/qe/bdd/CommonSteps.java:            fail("No build found for integration with name " + sanitizedName);
src/main/java/io/syndesis/qe/bdd/CommonSteps.java:        fail("Default PostgresDB connection has not been created, please contact engineering!");
src/main/java/io/syndesis/qe/bdd/CommonSteps.java:                fail("Todo app wasn't initilized in time");
src/main/java/io/syndesis/qe/bdd/CommonSteps.java:            fail("Waiting for Todo app was interrupted with exception: " + e.getMessage());
src/main/java/io/syndesis/qe/bdd/datamapper/AtlasMapperGenerator.java:            fail("Unable to map to \"ANY\" datashape!");
src/main/java/io/syndesis/qe/bdd/datamapper/AtlasMapperGenerator.java:            fail("Unable to find field with path " + mappingDef.getInputFields().get(0));
src/main/java/io/syndesis/qe/bdd/datamapper/AtlasMapperGenerator.java:            fail("Unable to find field with path " + mappingDef.getOutputFields().get(0));
src/main/java/io/syndesis/qe/servicenow/ServiceNow.java:                fail("Unable to unmarshall incident response", e);
src/main/java/io/syndesis/qe/servicenow/ServiceNow.java:            fail("Unable to convert incident to json", e);
src/main/java/io/syndesis/qe/wait/OpenShiftWaitUtils.java:            fail("Error thrown while checking if pod exists", e);
src/main/java/io/syndesis/qe/wait/OpenShiftWaitUtils.java:            fail("Error thrown while checking if pod exists", e);
src/main/java/io/syndesis/qe/templates/SyndesisTemplate.java:                fail("Unable to find image streams after " + maxRetries + " tries.");
src/main/java/io/syndesis/qe/templates/SyndesisTemplate.java:                    fail("Unable to import image for image stream " + is.getMetadata().getName() + " after 3 retries");
src/main/java/io/syndesis/qe/templates/SyndesisTemplate.java:                fail("Trying to deploy prod version using operator and system property " + TestConfiguration.PROD_REPOSITORY + " is not set!");
src/main/java/io/syndesis/qe/templates/SyndesisTemplate.java:                fail("Unable to find syndesis-server-config configmap after {} tries", maxRetries);
src/main/java/io/syndesis/qe/templates/SyndesisTemplate.java:                fail("Unable to process json", e);
src/main/java/io/syndesis/qe/templates/MysqlTemplate.java:            fail("MySQL database never started in pod.", e);
src/main/java/io/syndesis/qe/endpoints/ExtensionsEndpoint.java:            fail("Extension file not found", e);
src/main/java/io/syndesis/qe/endpoints/AtlasmapEndpoint.java:                fail("Bad datashape kind ({}) specified, only XML_SCHEMA, XML_INSTANCE, JSON_SCHEMA, JSON_INSTANCE are supported here", dsKind);
src/main/java/io/syndesis/qe/endpoints/TestSupport.java:        fail("Unable to successfully reset DB after 10 tries");
src/main/java/io/syndesis/qe/utils/GMailUtils.java:            fail("Exception thrown while tying to send an email.", e);
src/main/java/io/syndesis/qe/utils/GMailUtils.java:            fail("Exception was thrown while deleting messages.", e);
src/main/java/io/syndesis/qe/utils/ExportedIntegrationJSONUtil.java:            Assert.fail("Error while processing exported integration zip file");
src/main/java/io/syndesis/qe/utils/ExportedIntegrationJSONUtil.java:            Assert.fail("Error while processing exported integration zip entry");
src/main/java/io/syndesis/qe/utils/TelegramUtils.java:            fail("Api token for Telegram connector not found in credentials.json file!");
src/main/java/io/syndesis/qe/utils/TelegramUtils.java:            fail("Api token for Telegram connector not found in credentials.json file!");
src/main/java/io/syndesis/qe/utils/HttpUtils.java:            fail("Error while creating Http client", e);
src/main/java/io/syndesis/qe/utils/FtpUtils.java:            fail("Unable to delete file", path);
src/main/java/io/syndesis/qe/utils/FtpUtils.java:            fail("Unable to list files in FTP", ex);
src/main/java/io/syndesis/qe/utils/FtpUtils.java:            fail("Unable to upload test file: ", ex);
src/main/java/io/syndesis/qe/utils/FtpUtils.java:            fail("Unable to read FTP file " + directory + "/" + fileName);
src/main/java/io/syndesis/qe/utils/dballoc/DBAllocationManager.java:            Assertions.fail(e.getMessage());
src/main/java/io/syndesis/qe/utils/OpenShiftUtils.java:            fail("No pod found for pod name: " + podPartialName);
src/main/java/io/syndesis/qe/utils/OpenShiftUtils.java:                fail("Unable to use specified HTTP Method!");
src/main/java/io/syndesis/qe/utils/SampleDbConnectionManager.java:            fail("Error when handling external database", ex);
src/main/java/io/syndesis/qe/utils/SampleDbConnectionManager.java:            fail("Error creating DB connection.", e);
src/main/java/io/syndesis/qe/utils/FtpClientManager.java:        fail("Unable to create FTP client after 3 retries");
src/main/java/io/syndesis/qe/utils/mqtt/MqttUtils.java:            Assertions.fail("Sending a message should not have thrown any exception.");
src/main/java/io/syndesis/qe/utils/mqtt/MqttUtils.java:            Assertions.fail("There should be no error from Mqtt client!");

```